libp2p/crypto/p2pSigner_test.go

```diff
@@ -42,91 +81,138 @@ func TestP2pSigner_NewP2PSigner(t *testing.T) {
+	pkBytes, _ := pk.ToByteArray()
+	pubKey, _ := libp2pCrypto.UnmarshalSecp256k1PublicKey(pkBytes)
+	return pubKey
 }
 
 func TestP2pSigner_Verify(t *testing.T) {
 	t.Parallel()
 
-	sk := generatePrivateKey()
-	pk := sk.GetPublic()
 	payload := []byte("payload")
-	signer, _ := p2pCrypto.NewP2PSigner(sk)
-	libp2pPid, _ := peer.IDFromPublicKey(pk)
 
-	t.Run("invalid public key should error", func(t *testing.T) {
+	t.Run("fail to verify, should return err", func(t *testing.T) {
 		t.Parallel()
 
-		sig, err := signer.Sign(payload)
-		assert.Nil(t, err)
+		args := createDefaultP2PSignerArgs()
 
-		err = signer.Verify(payload, "invalid PK", sig)
-		assert.NotNil(t, err)
-		assert.Equal(t, "length greater than remaining number of bytes in buffer", err.Error())
-	})
-	t.Run("malformed signature header should error", func(t *testing.T) {
-		t.Parallel()
+		expectedErr := errors.New("expected error")
+		args.Signer = &mock.SingleSignerStub{
+			VerifyCalled: func(public crypto.PublicKey, msg, sig []byte) error {
+				return expectedErr
+			},
+		}
 
-		sig, err := signer.Sign(payload)
-		assert.Nil(t, err)
+		signer, _ := p2pCrypto.NewP2PSignerWrapper(args)
 
-		sig[0] = sig[0] ^ sig[1] ^ sig[2]
+		_, pk := generatePrivateKey()
+		peerID, err := peer.IDFromPublicKey(convertPublicKeyToP2PPublicKey(pk))
+		require.Nil(t, err)
 
-		err = signer.Verify(payload, core.PeerID(libp2pPid), sig)
-		assert.NotNil(t, err)
-		assert.Contains(t, err.Error(), "malformed signature: format has wrong type")
+		err = signer.Verify(payload, core.PeerID(peerID), []byte("sig"))
+		require.Equal(t, expectedErr, err)
 	})
-	t.Run("altered signature should error", func(t *testing.T) {
+
+	t.Run("should work", func(t *testing.T) {
 		t.Parallel()
 
-		sig, err := signer.Sign(payload)
-		assert.Nil(t, err)
+		args := createDefaultP2PSignerArgs()
 
-		sig[len(sig)-1] = sig[0] ^ sig[1] ^ sig[2]
+		verifyWasCalled := false
+		args.Signer = &mock.SingleSignerStub{
+			VerifyCalled: func(public crypto.PublicKey, msg, sig []byte) error {
+				verifyWasCalled = true
+				return nil
+			},
+		}
 
-		err = signer.Verify(payload, core.PeerID(libp2pPid), sig)
-		assert.Equal(t, crypto.ErrSigNotValid, err)
-	})
-	t.Run("sign and verify should work", func(t *testing.T) {
-		t.Parallel()
+		signer, _ := p2pCrypto.NewP2PSignerWrapper(args)
 
-		sig, err := signer.Sign(payload)
-		assert.Nil(t, err)
+		_, pk := generatePrivateKey()
+		peerID, err := peer.IDFromPublicKey(convertPublicKeyToP2PPublicKey(pk))
+		require.Nil(t, err)
 
-		err = signer.Verify(payload, core.PeerID(libp2pPid), sig)
-		assert.Nil(t, err)
+		err = signer.Verify(payload, core.PeerID(peerID), []byte("sig"))
+		require.Nil(t, err)
+
+		assert.True(t, verifyWasCalled)
 	})
 }
 
 func TestP2PSigner_SignUsingPrivateKey(t *testing.T) {
 	t.Parallel()
 
 	payload := []byte("payload")
+	pkBytes := []byte("private key bytes")
 
-	generator := p2pCrypto.NewIdentityGenerator()
```

Comment:
Please add a semi-integration test as the original test that you have changed. It is very important that this test exists & passes each time

Reply:
added semi integration test